op-batcher/batcher/driver.go

```diff
@@ -430,8 +430,14 @@ func (l *BatchSubmitter) mainLoop(ctx context.Context, receiptsCh chan txmgr.TxR
 				continue
 			}
 
+			// Lock the state to prevent any changes while we're computing sync actions
+			l.state.mu.Lock()
+			blocks := l.state.blocks
+			channels := l.state.channelQueue
+			l.state.mu.Unlock()
```

Comment:
Don't think this solves the race condition, since the  computeSyncActions function still calls functions on the channels. Here's what I had done: https://github.com/ethereum-optimism/optimism/pull/13169/files#diff-2a830eb26f5b3a174fa39ce7e267cbabe362ccb08b354c1b4e1abb9f60283808R115

Reply:
You might actually just want to make the `computeSyncActions` function a method on the channelManager?

Reply:
we've changed the locking to happen around the sync computation and pruning